src/shop/suppliers/base.py

```diff
@@ -32,3 +32,13 @@ def retrieve_product(self, sku):
         :type sku: str
         :rtype Union[SupplierProduct, None]
         """
+
+    @abstractmethod
+    def order_product(self, sku, qty):
+        """Places an order on product with given SKU.
```

Comment:
Well, that's a good justification, however it just saves readers trouble to write the full definition once at least.